### DISCUSSION TITLE
Loading Arnold Hydra Delegate In Custom C++ USD Based Application Failing.

### DISCUSSION THREAD

**Discussion by SimonCSmith** [Autodesk/arnold-usd#976] [Q&A]
I've built the Arnold Hydra Delegate from source (master branch) to use with a C++ USD (monolithic 21,11) based application (All source for USD and dependencies have also been build from scratch, all under Windows with Visual Studio 2019). 

For some reason when I try to use the Arnold Hydra Delegate it crashes out trying to create the plugin instance as the internal factory pointer is NULL. I can't for the life of me work out why this is as trying to work out how plugins get their factory is pretty hard going in the USD code. Embree, GL, and PRMan all work OK so i'm thinking this should be working and that it's a user error, but I can't work it out - I hope this is the right place to ask about it!

It loads the hdArnold.dll (and its dependencies) in the internal plugin->load() call quite happily, but increasing the reference count up from 0 fails due to this factory not being set up in the plugins registration (I assume). Can anyone help me out as to why this might occuring, or how to track down where the factory is set up so I can perhaps debug it to work out what the issue is.

**Comment by sebastienblor** [comment]
Hi @SimonCSmith , sorry for the lack of reply, we had missed this section for discussions.
Have you found a solution ? 
We're regularly building the render delegate for MayaUSD and Solaris, and they both use shared libraries (not monolithic). Would you have a way to try with shared libraries to see if this is causing the problem ?
Thanks ! (and sorry again for the delay)
